src/css/wizard-navigation-bar.scss

```diff
@@ -140,120 +120,155 @@ $wz-param-indicator-color: null !global;
+              background-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'background-color');
+            }
+          }
+
+          @else if $indicator-style == 'large-filled' {
+            @include wz-define-style($layout, $border-width: 0) {
+              background-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'background-color');
+            }
+          }
+
+          @else if $indicator-style == 'large-empty' {
+            @include wz-define-style($layout) {
+              $border-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'border-color');
+              @if $border-color != null {
+                // To optimize CSS output size, do not define any border styles when border-color is null
+                border: $wz-param-indicator-border-width solid $border-color;
+              }
+            }
+          }
+
+          @else if $indicator-style == 'large-filled-symbols' {
+            @include wz-define-style($layout, $border-width: 0) {
+              background-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'background-color');
+              color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'symbol-color');
+            }
+          }
+
+          @else if $indicator-style == 'large-empty-symbols' {
+            @include wz-define-style($layout) {
+              $border-color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'border-color');
+              @if $border-color != null {
+                // To optimize CSS output size, do not define any border styles when border-color is null
+                border: $wz-param-indicator-border-width solid $border-color;
+              }
+              color: wz-get-color($indicator-style, $wz-param-step-state, $wz-param-indicator-state, 'symbol-color');
+            }
+          }
+        }
+      }
     }
   }
 }
 
 
-// Helper mixin to define indicator styles for 'horizontal' and 'vertical' layouts
-@mixin wz-define-layout-styles() {
-  @each $layout in 'horizontal', 'vertical' {
-    aw-wizard-navigation-bar.#{$layout} {
-      @include wz-define-styles($layout);
+// Retrieve color from the color definitions map ($wz-colors)
+//
+// Arguments:
+//   $indicator-style: Indicator style: 'small', 'large-filled' etc.
+//   $step-state:      Wizard step state: small', 'large-filled' etc.
+//   $indicator-state: Indicator state: 'default' or 'hover'
+//   $property:        Property to retrieve: 'background-color', 'border-color', 'symbol-color'
+//
+// Returns:
+//   Color value like #E6E6E6 or `null` if
+//     - the requested property is not specified in the color definitions map
+//     - the corresponding color definitions map entry is explicit `null`
+//
+@function wz-get-color($indicator-style, $step-state, $indicator-state, $property) {
+  $maps: (
```

Comment:
This is the first time I've seen a map in `scss`. Can you elaborate a bit more about what `$maps` contains after defining it? And maybe also what exactly you access in the `$each` loop?

Reply:
Yes, I added comments in 328fa26